### PR TITLE
hcm: finalize split between ActiveStream and FilterManager

### PR DESCRIPTION
**Pull request by snowp**
This moves the remaining functionality in ActiveStream into the FilterManager, which in turn means that FM no longer needs to be a `friend` of AS.

Signed-off-by: Snow Pettersen <aickck@gmail.com>

Risk Level: Medium
Testing: Existing tests
Docs Changes: n/a
Release Notes: n/a
Part of #10454

### CODE REVIEW

File: source/common/http/conn_manager_impl.cc [hunk 1]
```diff
@@ -2881,17 +2914,25 @@ void ConnectionManagerImpl::ActiveStreamEncoderFilter::responseDataTooLarge() {
   }
 }
 
+void ConnectionManagerImpl::ActiveStream::onResponseDataTooLarge() {
+  connection_manager_.stats_.named_.rs_too_large_.inc();
```
Comment (alyssawilk) [inline source/common/http/conn_manager_impl.cc:2918]: Do we have a TODO for how to more cleanly handles stats client and server side?
seems like the manager could do stats if it had pointers to them.

File: source/common/http/conn_manager_impl.h [hunk 2]
```diff
@@ -481,7 +573,7 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
       const char* spaces = spacesForLevel(indent_level);
       os << spaces << "FilterManager " << this << DUMP_MEMBER(state_.has_continue_headers_)
          << DUMP_MEMBER(state_.decoding_headers_only_) << DUMP_MEMBER(state_.encoding_headers_only_)
-         << "\n";
+         << DUMP_MEMBER(state_.encoding_headers_only_) << "\n";
```
Comment (alyssawilk) [inline source/common/http/conn_manager_impl.h:576]: twice?

File: source/common/http/conn_manager_impl.h [hunk 3]
```diff
@@ -963,11 +1083,11 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     }
 
     ConnectionManagerImpl& connection_manager_;
+    const uint64_t stream_id_;
```
Comment (alyssawilk) [inline source/common/http/conn_manager_impl.h:1086]: looks like this is in 2 places.  candidate for future clean up?

File: source/common/http/conn_manager_impl.cc (hunk 1 above)
Comment (snowp) [reply source/common/http/conn_manager_impl.cc:2918]: No TODO yet, I'll add one. I'm a bit unsure whether this should be callbacks, required counters or optional counters (if we require counters we'll need to add stats for this for upstream calls, which would add more default stats etc.)

File: source/common/http/conn_manager_impl.h (hunk 2 above)
Comment (snowp) [reply source/common/http/conn_manager_impl.h:576]: oops!

File: source/common/http/conn_manager_impl.h (hunk 3 above)
Comment (snowp) [reply source/common/http/conn_manager_impl.h:1086]: Conceptually it feels like this belongs on the ActiveStream, but at this point the main use case of it in AS is for `ENVOY_STREAM_LOG`. I guess one option would be to have the AS allocate the stream id and pass it to the FM, then refer to the FM when doing stream logging from AS?

Comment (alyssawilk) [reply source/common/http/conn_manager_impl.h:1090]: I was just thinking one could snag it from the other via accessor / callback.
Take a look and if you think it makes sense you can do in a follow-up?

Comment (snowp) [reply source/common/http/conn_manager_impl.h:1090]: Yeah I'll see if I can find something that makes sense

**Pull request merged**